compliance/collection/nodescanv2/nodescan.go

```diff
@@ -16,5 +19,60 @@ type NodeScan struct {
 
 // Scan scans the current node and returns the results as storage.NodeInventory object
 func (n *NodeScan) Scan(nodeName string) (*storage.NodeInventory, error) {
-	return nil, errors.New("Not implemented")
+	componentsHost, err := nodes.Analyze(nodeName, "/host/", false)
+	log.Info("Finished node inventory /host scan")
+	if err != nil {
+		log.Errorf("Error scanning node /host inventory: %v", err)
+	}
+	log.Infof("Components found under /host: %v", componentsHost)
+	if err != nil {
+		return nil, err
+	}
+
+	var protoComponents *scannerV1.Components
+	if componentsHost != nil {
+		protoComponents = protoComponentsFromScanComponents(componentsHost)
+	}
+	m := &storage.NodeInventory{
+		NodeName:   nodeName,
+		ScanTime:   timestamp.TimestampNow(),
+		Components: protoComponents,
+	}
+	return m, nil
+}
+
+func protoComponentsFromScanComponents(c *nodes.Components) *scannerV1.Components {
+	var components []*scannerV1.RHELComponent
+	// For now, we only care about RHEL components, but this must be extended once we support non-RHCOS
+	if c.CertifiedRHELComponents != nil {
+		components = convertRHELComponents(c.CertifiedRHELComponents)
+	}
+	pc := scannerV1.Components{
+		Namespace:          c.OSNamespace.Name,
+		OsComponents:       nil,
+		RhelComponents:     components,
+		LanguageComponents: nil,
+	}
+	return &pc
+}
+
+func convertRHELComponents(rc *database.RHELv2Components) []*scannerV1.RHELComponent {
+	v1rhelc := make([]*scannerV1.RHELComponent, 0)
+	if rc.Packages == nil {
+		log.Warn("No RHEL packages found in scan result")
+		return nil
+	}
+	for _, rhelc := range rc.Packages {
+		v1rhelc = append(v1rhelc, &scannerV1.RHELComponent{
+			Id:          0,
```

Comment:
Would `Id == 0` for each `RHELComponent` be a problem?

Reply:
I have investigated this, thank you for pointing it out.
In the fake message, we don't set the ID field at all and the message is correctly processed by Scanner according to reports from Piotr. So for now I have removed the ID field completely.

I would like to address this together with the question of datatype converters with Ross and Victor, to get some insight how we can best proceed here.